operator/gitops/argocd/pipeline-service/openshift-pipelines/chains-secrets-config.yaml

```
@@ -61,37 +61,43 @@ spec:
             - /bin/bash
             - -c
             - |
+              set -o errexit
+              set -o nounset
+              set -o pipefail
+
+              namespace="openshift-pipelines"
+              secret="signing-secrets"
+
               cd /tmp
-              # Try to handle that nicely. The object is expected to always exist so check the data.
-              SIG_KEY_DATA=$(kubectl get secret signing-secrets -n openshift-pipelines -o jsonpath='{.data}')
-              if [[ -n $SIG_KEY_DATA ]]; then
+
+              if [ "$(kubectl get secret "$secret" -n "$namespace" -o jsonpath='{.data}')" != "" ]; then
                 echo "Signing secret exists and is non-empty."
               else
                 # Delete secret/signing-secrets if already exists since by default cosign creates immutable secrets
-                kubectl delete secrets -n openshift-pipelines signing-secrets --ignore-not-found=true
+                kubectl delete secrets "$secret" -n "$namespace" --ignore-not-found=true
 
                 # To make this run conveniently without user input let's create a random password
                 RANDOM_PASS=$( head -c 12 /dev/urandom | base64 )
 
                 # Generate the key pair secret directly in the cluster.
                 # The secret should be created as immutable.
-                echo "Generating k8s secret/signing-secrets with key-pair"
-                env COSIGN_PASSWORD=$RANDOM_PASS cosign generate-key-pair k8s://openshift-pipelines/signing-secrets
+                echo "Generating k8s secret/$secret in $namespace with key-pair"
+                env COSIGN_PASSWORD=$RANDOM_PASS cosign generate-key-pair "k8s://$namespace/$secret"
               fi
 
               # If the secret is not marked as immutable, make it so.
-              if [ "$(kubectl get secret signing-secrets -n openshift-pipelines -o jsonpath='{.immutable}')" != "true" ]; then
+              if [ "$(kubectl get secret "$secret" -n "$namespace" -o jsonpath='{.immutable}')" != "true" ]; then
                 echo "Making secret immutable"
-                kubectl get secret signing-secrets -n openshift-pipelines -o yaml \
+                kubectl get secret "$secret" -n "$namespace" -o yaml \
                 | yq eval '.immutable = true' - \
                 | kubectl apply -f -
```

Using kubectl patch will remove the dependency on `yq`
```suggestion
                kubectl patch secret "$secret" -n "$namespace" --dry-run=client --patch='{"immutable": true}' -o yaml \
                | kubectl apply -f -
```

---

Done